src/Action.hs

```diff
@@ -67,8 +70,12 @@ data UserState
 
 makeLenses ''UserState
 
+-- TODO: here @ActionPersist Persist@ is hardwired, so either
+-- parameterize ActionM by the persistance implementation
+-- or remove ActionPersist from ActionM and add by hand elsewhere
+-- (messy and lots of places need to be touched).
 class ( ActionLog m
-      , ActionPersist m
+      , ActionPersist Persist m
```

Comment:
I am ok with the further parameterizing of the ActionM monad. I had the same idea, for me it clearly says that for any Action we implement has to be independent of which persistent implemantation we use.